source/NanostackRfPhyAtmel.cpp

```diff
@@ -2200,21 +2214,30 @@ int8_t NanostackRfPhyAtmel::rf_register()
     // Read the mac address if it hasn't been set by a user
     rf = _rf;
     test_pins = _test_pins;
+    se2435_pa_pins = _se2435_pa_pins;
     if (!_mac_set) {
+// Unless AT24MAC is available, using randomly generated MAC address
+#if !defined(DISABLE_AT24MAC)
         int ret = _mac.read_eui64((void *)_mac_addr);
         if (ret < 0) {
             rf = NULL;
             rf_if_unlock();
             return -1;
         }
+#else
+        randLIB_seed_random();
```

Comment:
Are we randomise MAC at driver side in earlier with s2lp?

Reply:
Yes, and if it doesn't, eventually Stack will randomize it.

Reply:
Do you need to check that there is no value before randomising or is this always output?

Reply:
"if (!_mac_set)" on line 2218 should take care that MAC address is randomised only once if for some reason application wants to register same driver multiple times. Why would it? I'm not sure though who is setting _mac_set to true.